sympy/printing/smtlib.py

```diff
@@ -338,7 +338,7 @@ def smtlib_code(
                 sym not in p._known_constants and
                 sym not in p.symbol_table
             ):
-                log_warn(f"Could not infer type of `{sym}`. Defaulting to float.")
+                warnings.warn(f"Could not infer type of `{sym}`. Defaulting to float.")
                 p.symbol_table[sym] = float
```

Comment:
Why not just raise an error here?

Reply:
Because I don't think these qualify as program-crashing errors. I speculate that a large majority of sympy queries are going to be over real numbers, so it would be excessively pedantic and cumbersome to require users to always specify that.

I am trying to serve 2 groups of (imagined) users:
1. Many users just want a more powerful solver that can handle constraints. They don't want to have to deal with the specifics of SMT and don't want to have to write-out any more information to the SMT solver than would be required for a regular sympy `solve()` call.
2. There are also more advanced users, specifically from academia, who care deeply about the specifics of what the solver is acting on. These users care about being precise because they are trying to construct robust mathematical _proofs_ (not just trying to solve an equation). They will likely make full use of `symbol_table`, `known_functions`, etc. and would want a very explicit API which warns loudly of any ambiguity in their query.

Making these errors would hurt group 1 and making these silent would hurt group 2. Thus I decide that warnings are the best way forward.

Reply:
Issuing warnings when something is used in the way that it is expected to be used is just bad interface design IMO.

Why not just document that if `symbol_table` is not passed then all symbols will be assumed to be floats (since I presume that's the behaviour here)? Then if `symbol_table` is passed but doesn't contain the symbol an exception can be raised.